framework/e2e/test_control_api.sh

```diff
@@ -112,21 +112,22 @@ cleanup_and_exit() {
     exit $1
 }
 
-# Check for "Run finished" in a loop with a timeout
 while [ "$found_success" = false ] && [ $elapsed -lt $timeout ]; do
-    if grep -q "ERROR" flwr_output.log; then
-        echo "An ERROR occurred during training. Exiting."
-        cleanup_and_exit 1
-    elif grep -q "Run finished" flwr_output.log; then
-        echo "Training worked correctly!"
-        found_success=true
-        cleanup_and_exit 0
+    # Run the command and capture output
+    output=$(flwr ls . e2e --format=json)
+
+    # Extract status from the first run (or loop over all if needed)
+    status=$(echo "$output" | jq -r '.runs[0].status')
+
+    echo "Current status: $status"
+
+    if [ "$status" == "finished:completed" ]; then
+      found_success=true
+      echo "Training worked correctly!"
```

Comment:
```suggestion
      echo "Training worked correctly!"
      cleanup_and_exit 0
```
I think we should cleanup here too.